lib/cisco_node_utils/bridge_domain.rb

```diff
@@ -124,11 +124,17 @@ def destroy
     #                      PROPERTIES                      #
     ########################################################
 
-    # Bridge-Domain name assigning case
-    # bridge-domain 100
-    #   name bd100
     def bd_name
-      config_get('bridge_domain', 'bd_name', bd: @bd_ids)
+      match = config_get('bridge_domain', 'bd_name', bd: @bd_ids)
```

Comment:
The getters for `bd_name` and `fabric_control` can be significantly simplified by using structured output.

For example:
```yaml
bd_name:
  kind: string
  get_data_format: nxapi_structured
  get_command: "show bridge-domain <bd>"
  get_context: ["TABLE_bd", "ROW_bd"]
  get_value: ["bd-id <bd>", "bd-name"]
  set_value: "<state> name <name>"
  default_value: ~
```

```ruby
def bd_name
  config_get('bridge_domain', 'bd_name', bd: @bd_ids)
end
```

Reply:
**Note:** This will cause two of the tests to fail with the following:

```shell
Node under test:
  - name  - n7k-j
  - type  - N7K-C7009
  - image - bootflash:///n7000-s2-dk9.7.3.0.D1.1.bin

TestBridgeDomain#test_bd_name = 5.96 s = F

Finished in 7.861067s, 0.1272 runs/s, 0.1272 assertions/s.

  1) Failure:
TestBridgeDomain#test_bd_name [tests/test_bridge_domain.rb:109]:
Error: Bridge-Domain name not initialized to default.
Expected: nil
  Actual: "Bridge-Domain101"
```

This needs to be fixed.  We should not be checking for `nil` here but rather the default name.

Reply:
I still did a small processing inside the object file rather than fixing the test code. The reason is that the default value is nil in the old code and this could create a backward compatibility issue. The other reason is that the return value is either the set value or "Bridge-domain<bd>". So the get will always return the "Bridge-domain<bd" if the name is not set. This can cause problems in the puppet module side as the default value is still nil there. So idempotent issues will arise unless we fix the provider code also. I don't see value in changing the code in multiple places. So just returning nil if the get value is "Bridge-domain<bd" is simple one line change and will work just like before.

Reply:
I suppose it's better to normalize to something we control rather then let the box decide in this case.

Reply:
Actually, had a chance to think about this some more.  I still believe we should change the default behavior even if it means changing puppet.  Here is my rational.

1. There likely won't be a backwards compatibility issue because we see the same default `bd_name` behavior in helsinki and beverly_hills.
2. Even if the default name format changes, it's still a default name and the actual name of the bridge-domain so we should return that as the value.
3. If users try to use the default name as the actual name in the manifest, it will get returned as nil.